### PR TITLE
Log Gradient Norms

### PR DESCRIPTION
**Pull request by maanug-nv**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/strategies.py [hunk 1]
```diff
@@ -491,6 +505,22 @@ def training_step(self, dataloader_iter, *args: Any, **kwargs: Any) -> STEP_OUTP
...
+    def optimizer_step(
+        self,
+        optimizer: torch.optim.Optimizer,
+        closure: Callable[[], Any],
+        model: Optional[Union["pl.LightningModule", nn.Module]] = None,
+        **kwargs: Any,
+    ) -> Any:
+        optimizer_output = super().optimizer_step(optimizer, closure, model, **kwargs)
+
+        if isinstance(optimizer, McoreDistributedOptimizer):
```
Comment (maanug-nv) [inline nemo/lightning/pytorch/strategies.py:518]: Currently will only work for MCore Distributed Optimizers. I'm open to making this more general by returning a dictionary from `McoreDistributedOptimizer.step()`. I don't know if PTL does anything with the outputs of the optimizer step.

Comment (maanug-nv) [reply nemo/lightning/pytorch/strategies.py:518]: @marcromeyn ^^ for the design choice

File: nemo/lightning/pytorch/strategies.py [hunk 2]
```diff
@@ -491,6 +505,22 @@ def training_step(self, dataloader_iter, *args: Any, **kwargs: Any) -> STEP_OUTP
...
+            _, grad_norm = optimizer_output
+            self.lightning_module.log('grad_norm', grad_norm, batch_size=1)
+
+        return optimizer_output
```
Comment (akoumpa) [inline nemo/lightning/pytorch/strategies.py:522]: `return optimizer_output` or `return optimizer_output[0]` ?

Comment (maanug-nv) [reply nemo/lightning/pytorch/strategies.py:522]: Yeah, this was my question regarding if it makes any difference. not sure if PTL cares about the output of this. I can make the change to be safe.

**Review by akoumpa — APPROVED** [review]
LGTM, thanks.

**Review by marcromeyn — APPROVED** [review]
